add frozon_numpy as a builtin library to torch::deploy

shunting314 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #66297

Link register_numpy.cpp with the embedded interpreter will register numpy as a builtin library.

Differential Revision: [D31490434](https://our.internmc.facebook.com/intern/diff/D31490434/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31490434/)!

shunting314 committed:
add frozon_numpy as a builtin library to torch::deploy

Link register_numpy.cpp with the embedded interpreter will register numpy as a builtin library.

Differential Revision: [D31490434](https://our.internmc.facebook.com/intern/diff/D31490434/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31490434/)!

[ghstack-poisoned]

shunting314 committed:
Update on "add frozon_numpy as a builtin library to torch::deploy"


Link register_numpy.cpp with the embedded interpreter will register numpy as a builtin library.

Differential Revision: [D31490434](https://our.internmc.facebook.com/intern/diff/D31490434/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31490434/)!

[ghstack-poisoned]

shunting314 committed:
Update on "add frozon_numpy as a builtin library to torch::deploy"


Link register_numpy.cpp with the embedded interpreter will register numpy as a builtin library.

Differential Revision: [D31490434](https://our.internmc.facebook.com/intern/diff/D31490434/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31490434/)!

[ghstack-poisoned]

shunting314 committed:
Update on "add frozon_numpy as a builtin library to torch::deploy"


Link register_numpy.cpp with the embedded interpreter will register numpy as a builtin library.

Differential Revision: [D31490434](https://our.internmc.facebook.com/intern/diff/D31490434/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31490434/)!

[ghstack-poisoned]

shunting314 committed:
Update on "add frozon_numpy as a builtin library to torch::deploy"


Link register_numpy.cpp with the embedded interpreter will register numpy as a builtin library.

Differential Revision: [D31490434](https://our.internmc.facebook.com/intern/diff/D31490434/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31490434/)!

[ghstack-poisoned]

suo reviewed (approved):
nice, really like how clean this registration is overall!

shunting314 committed:
Update on "add frozon_numpy as a builtin library to torch::deploy"


Link register_numpy.cpp with the embedded interpreter will register numpy as a builtin library.

Differential Revision: [D31490434](https://our.internmc.facebook.com/intern/diff/D31490434/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D31490434/)!

[ghstack-poisoned]
